packages/core/src/services/documents/recomputeChanges/getHeadDocumentsAndDraftDocuments.ts

```diff
@@ -0,0 +1,81 @@
+import { Commit } from '$core/browser'
+import { database } from '$core/client'
+import { Result } from '$core/lib'
+import { buildCommitsScope } from '$core/repositories/commitsRepository/utils/buildCommitsScope'
+import { getHeadCommitForProject } from '$core/repositories/commitsRepository/utils/getHeadCommit'
+import { DocumentVersionsRepository } from '$core/repositories/documentVersionsRepository'
+import { ProjectsRepository } from '$core/repositories/projectsRepository'
+
+async function getProjectFromCommit(
+  { commit, workspaceId }: { commit: Commit; workspaceId: number },
+  tx = database,
+) {
+  const projectsScope = new ProjectsRepository(workspaceId, tx)
+  const projectResult = await projectsScope.getProjectById(commit.projectId)
+
+  if (projectResult.error) return projectResult
+
+  return Result.ok(projectResult.value)
+}
+async function getDraftDocuments(
+  { commit, workspaceId }: { commit: Commit; workspaceId: number },
+  tx = database,
+) {
+  const docsScope = new DocumentVersionsRepository(workspaceId, tx)
+  const draftChangesResult = await docsScope.listCommitChanges(commit)
+  if (draftChangesResult.error) return Result.error(draftChangesResult.error)
+  return Result.ok(draftChangesResult.value)
+}
+
+async function getDocumentsAtCommit(
+  { commit, workspaceId }: { commit: Commit; workspaceId: number },
+  tx = database,
+) {
+  const projectResult = await getProjectFromCommit({ commit, workspaceId }, tx)
+  if (projectResult.error) return projectResult
+
+  const commitsScope = buildCommitsScope(workspaceId, tx)
```

Comment:
Decouple commits scope from CommitsRepository. 
This file is used inside `recomputeChanges` that is used in `CommitsRepository` resulting in Circular dependency warnings in Rollup ♻️